### PR TITLE
WINC-677: [e2e] Modularize linux curler job creation

### PR DESCRIPTION
**Pull request by saifshaikh48**
This PR consolidates the creation of the linux curl commands in the e2e tests so that there is a standard way of calling the curler jobs.

### CODE REVIEW

**Comment by saifshaikh48** [comment]
/test aws-e2e-upgrade

cluster install failure

**Comment by alinaryan** [comment]
Thanks for working on this, 
/lgtm

File: test/e2e/network_test.go
Comment (aravindhp) [inline test/e2e/network_test.go:579]: I suggest making the signature much simpler i.e. combine endpoint, portNum and portName into a URL and construct that at the call site. WDYT?

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @saifshaikh48

File: test/e2e/network_test.go
Comment (aravindhp) [inline test/e2e/network_test.go:594]: Why do we need to do curl twice here? I think this can be collapsed into one.

Comment (saifshaikh48) [reply test/e2e/network_test.go:587]: Not sure, let me try to rework it

Comment (aravindhp) [reply test/e2e/network_test.go:587]: This code is pretty convoluted and can be simplified 😄

File: test/e2e/network_test.go
Comment (aravindhp) [inline test/e2e/network_test.go:584]: You are still calling `curl` twice here. Can't you use `break` or `continue`?

File: test/e2e/network_test.go
Comment (saifshaikh48) [reply test/e2e/network_test.go]: @aravindhp After a bit I was able to see the purpose of it. I added some docs around it

File: test/e2e/network_test.go
Comment (aravindhp) [reply test/e2e/network_test.go:584]: In other words, always have the [while loop](https://linuxize.com/post/bash-while-loop/).

File: test/e2e/network_test.go
Comment (aravindhp) [reply test/e2e/network_test.go]: I still don't see why it needs to be this way. If we are trying to avoid flakes, we just retry the curl command N number of times and exit with an error.

File: test/e2e/network_test.go
Comment (saifshaikh48) [reply test/e2e/network_test.go:584]: the original intent of the 2 curls was to error on the second failure. I'll see if a do-while loop will allow me to remove the second one

Comment (aravindhp) [reply test/e2e/network_test.go:584]: You can use [this](https://www.networkworld.com/article/3541298/how-to-repeat-a-linux-command-until-it-succeeds.html) for inspiration.

Comment (saifshaikh48) [reply test/e2e/network_test.go:584]: After exploring this further I lean towards the current approach. To make it only 1 curl call we'd have to introduce a counter and it looks more complex than the current code.

Comment (aravindhp) [reply test/e2e/network_test.go:584]: Yeah I was thinking the counter would be more clear but I will defer to your judgement on this. Do we really need the continuous option? Won't doing the loop work in all cases?

**Review by aravindhp — APPROVED** [review]
Thanks for working on this, @saifshaikh48

**Comment by mansikulkarni96** [comment]
LGTM

**Comment by saifshaikh48** [comment]
/hold

depends on https://github.com/openshift/windows-machine-config-operator/pull/1372 to unblock CI

**Comment by saifshaikh48** [comment]
/retest-required

AWS cluster install failures

**Pull request merged**
